rle/rleplus.go

```diff
@@ -75,42 +75,54 @@ func (rle *RLE) MarshalJSON() ([]byte, error) {
 		return nil, err
 	}
 
-	var ret []uint64
+	count, err := rle.Count()
+	if err != nil {
+		return nil, err
+	}
+
+	var ret = struct {
```

Comment:
this struct should be extracted so it isn't duped in the marshaler and unmarshaler (and the test)

Reply:
How about de-duping the usage in the un/marshallers but leaving the anonymous struct in the test? Propose this as deduping everywhere will require exporting the struct, wish is undesirable since it should only be used internally.